src/Microsoft.Azure.SignalR.SignalRService/ServiceManager.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Azure.SignalR.SignalRService
+{
+    public class ServiceManager
+    {
+        public ServiceHubContext CreateHubContextAsync(string hubName)
+        {
+            throw new NotImplementedException();
+        }
+
+        public string GenerateAccessToken(string audience, TimeSpan? lifeTime = null)
```

Comment:
It's better to not take audience as input, user may not have enough knowledge about how to compose an audience.

Reply:
Ok.

How about taking service endpoint, scenario and other parameters as input. Audience is composed in `GenerateAccessToken`



``` C#
    // Scenario.cs

    public enum Scenario
    {
        Broadcast,
        SendToUser,
        SendToGroup,
        AddUserToGroup,
        RemoveUserFromGroup
    }

```


``` C#
// Take connection Id for placeholder for now, the SDK may support sending messages to some connection via connection Id
public string GenerateAccessToken(string endpoint, Scenario scenario, string hubName, string connectionId = null, string userId = null, string groupName = null, TimeSpan? lifeTime = null)


```

Reply:
Also need one for client connection

Reply:
Personally I'd recommend 'scope' over 'scenario'

Reply:
We should also consider to support bigger scope like 'server' which allows all kinds of server operation

Reply:
> Also need one for client connection 

Add `Service`, `Hub`, `SendToClient`, `AddConnectionToGroup` and `RemoveConnectionFromGroup`
``` C#
    public enum Scope
    {
        Service,
        Hub,
        Broadcast,
        SendToUser,
        SendToClient,
        SendToGroup,
        AddUserToGroup,
        RemoveUserFromGroup,
        AddConnectionToGroup,
        RemoveConnectionFromGroup
    }
```
> Personally I'd recommend 'scope' over 'scenario'

Ok, `Scope` sounds is more common in auth.


> We should also consider to support bigger scope like 'server' which allows all kinds of server operation

Do you mean the scope for hub? The largest scope for `ServiceHubContext` is `https://<instance-name>.service.signalr.net/api/v1/hubs/<hub-name>/` . Also discussed with @vwxyzh , we will support both bigger scope for the hub and smaller scope (current finer scope for groups/user/...).

Do we need the scope for service. For now, `ServiceHubContext` doesn't need it, but future libraries may need it.

Reply:
For now, I think at least we need one for client (token for client to connect to a hub) and one for server (token for all server side operations on a hub like broadcast and send)

Reply:
We consider that this library supports scope for hub level for now. Extending levels for scope in the future. If you have any requirement about this, please feel free to let me know.

Reply:
Delete parameter endpoint in `GenerateAccessToken`,  that is `string GenerateAccessToken(Scope scope, string hubName, string connectionId = null, string userId = null, string groupName = null, TimeSpan? lifeTime = null)`, since service manager hold connection string, which contains endpoint.

Reply:
Discussed with @chenkennt, we consider using connection string directly for server side usage and consider generate different connection string for different permissions. We only generate client access token only. Therefore remove this one.